snakemake/io.py

```diff
@@ -1436,13 +1436,43 @@ def get_func_ret(func):
         return do_expand(wildcard_values)
 
 
-def multiext(prefix, *extensions):
+@dataclass
+class MultiextValue:
+    prefix: str
+    name: Optional[str] = None
+
+
+def multiext(prefix, *extensions, **named_extensions):
     """Expand a given prefix with multiple extensions (e.g. .txt, .csv, _peaks.bed, ...)."""
     if any((r"/" in ext or r"\\" in ext) for ext in extensions):
         raise WorkflowError(
             r"Extensions for multiext may not contain path delimiters (/,\)."
         )
-    return [flag(prefix + ext, "multiext", flag_value=prefix) for ext in extensions]
+    if any((r"/" in ext or r"\\" in ext) for ext in named_extensions.values()):
+        raise WorkflowError(
+            r"Extensions for multiext may not contain path delimiters (/,\)."
+        )
```

Comment:
Here, you could just use itertools.chain to evaluate both extensions and named_extensions without code duplication.